mitodl/mitxpro
# feat: Add "sign up for more information" on the xPRO product pages

### What are the relevant tickets?
[#3591](https://github.com/mitodl/hq/issues/3591)

### Description (What does it do?)
- This PR adds "sign up for more information" [feature](https://github.com/mitodl/hq/issues/3591) on the xPRO product pages(for both internal/external courses or programs). 
- The implementation will enable the CMS to enter a form ID for a course or program page, which will then be used to display the form on the 'Keep me updated' CTA button for the respective course or program.

**Note:**
~~This PR is currently in progress for a few changes to be incorporated after getting [feedback](https://github.com/mitodl/mitxpro/pull/2906#issuecomment-1967538372).~~

### How can this be tested?

- Switch to this branch, then execute migrations. This action will add the `marketing_hubspot_form_id` field into the course/program CMS page.
- Navigate to any course/program detail page. The `Keep me updated` button should remain invisible since the `marketing_hubspot_form_id` is currently null.
- Proceed to create various HubSpot forms, similar to those listed in the vendor (or Xpro) [form list](https://docs.google.com/spreadsheets/d/14iYQ61sktb1At32b9-KrO0yk8DZoaB4Nc98U04eUdyo/edit?pli=1#gid=838625130). Assign the respective form IDs to the `marketing_hubspot_form_id` field in the course/program CMS page.
- Verify whether the 'Keep me updated' CTA button appears on the course/program details page when the form ID is present in the corresponding course/program cms`marketing_hubspot_form_id` field.  **Note:** The 'Keep me updated' button remains invisible if the user is already enrolled in the course/program.
   - Make sure, the HubSpot portal ID associated with the form matches your local environment for it to function properly.
- Check the form loading, submission, cancellation, and confirmation messages are working properly.
- Ensure that this feature works for both internal and external courses/programs.

**Note:** The 'Keep me updated' feature will only be visible on the course/program detail page if the form ID is present in the marketing_hubspot_form_id field of the course/program CMS page, and the user is not enrolled in the course/program.

### Screenshots (if appropriate):
Attached screenshots demonstrate the current progress.


CTAs(`keep me updated` buttons on hero and subnav):
<img width="1280" alt="signup-1" src="https://github.com/mitodl/mitxpro/assets/41048311/8f3f8389-e01b-41e2-ab2f-68193616d80e">

Form:
<img width="1279" alt="signup 2" src="https://github.com/mitodl/mitxpro/assets/41048311/acfe2438-73da-40d6-988b-a66dbe149daa">

Thankyou/Confirmation Message([updated](https://github.com/mitodl/mitxpro/pull/2906#issuecomment-1988426352)):
<img width="1440" alt="signup-3 - removed close button" src="https://github.com/mitodl/mitxpro/assets/41048311/f78a0810-df3f-4f90-9673-abe390cfbb9d">



Mobile:
<img width="894" alt="hubspot-form-mobile-view" src="https://github.com/mitodl/mitxpro/assets/41048311/4347adc5-a284-4ced-84bf-0cd5d098a384">

### Additional Context
